provisioner/template.go

```diff
@@ -91,10 +91,14 @@ func renderTemplate(context *templateContext, file string) (string, error) {
 		"sha256": func(value string) (string, error) {
 			return fmt.Sprintf("%x", sha256.Sum256([]byte(value))), nil
 		},
-		"asgSize":              asgSize,
-		"azID":                 azID,
-		"azCount":              azCount,
-		"split":                split,
+		"asgSize": asgSize,
+		"azID":    azID,
+		"azCount": azCount,
+		"split":   split,
+		"join":    join,
+		"stringSlice": func(args ...string) []string {
+			return args
+		},
```

Comment:
I think we can use existing list for the purpose, can't we?

Reply:
The problem is `list` produces `[]interface{}` and `join` takes `[]string`. Making `join` take `[]interface{}` seem ugly.

Reply:
> Making join take []interface{} seem ugly.

sprig does that though https://github.com/Masterminds/sprig/blob/fc7fc0d6a0377bca7049c4a99e80b85f222d8caf/strings.go#L199-L201

Maybe we can just use it (like we use `indent`) - it allowes piped input and takes case of typo conversion...

Reply:
Another option is to add `toStrings` helper
https://github.com/Masterminds/sprig/blob/fc7fc0d6a0377bca7049c4a99e80b85f222d8caf/functions.go#L186
```
{{ list "foo" "bar" | toStrings | join "," }}
```

Reply:
> stringSlice

If we go this way then lets name it `stringList` or `slist` to make it aligned with existing `list`.

Reply:
Ok, then let's just do the sprig `join` function. I have changed it to use that one.